app.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'student'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'All People:'
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person(name, age, type, specialization = nil, classroom = nil)
+    if type == 'teacher'
+      person = Teacher.new(age, specialization, name: name)
+    elsif type == 'student'
+      person = Student.new(age, classroom, name: name)
+    else
+      puts 'Invalid person type!'
+      return
+    end
+
+    @people << person
+    puts "Person created successfully. ID: #{person.id}"
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def create_rental(person_id, book_title, rental_date)
+    person = @people.find { |p| p.id == person_id }
+    book = @books.find { |b| b.title == book_title }
+
+    if person.nil?
+      puts 'Person not found!'
```

Comment:
Kindly make sure that when creating a rental a user can select a person from the person's list and select a book from the existing book list to create a rental.

- Currently, your options are to add a user with an ID and enter the book name manually. This is not an effective way since a user can enter a book name that doesn't exist in the booklist causing inconsistent data in your application.

Rental creation in the example video:

![image](https://github.com/karanJ2212/School_library_ruby/assets/38088818/d3f0d80f-0faa-4a04-ad50-7bca1ae5d035)

Rental creation in the app:

![image](https://github.com/karanJ2212/School_library_ruby/assets/38088818/8db60caf-0272-4ed5-ac65-80f3daab3031)

- Kindly make sure to also fix the error and make sure that rentals can be created successfully. This is because your add rentals method is named `rent` yet you are calling the method `rentals` in the Rental class for the Book and Person associations which do not exist.